contracts/test/TestWarmColdAccount.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.12;
+/* solhint-disable no-inline-assembly */
+
+import "../interfaces/IAggregatedAccount.sol";
+import "../interfaces/IEntryPoint.sol";
+
+
+contract TestWarmColdAccount is IAggregatedAccount {
+    IEntryPoint private ep;
+    uint public state = 1;
+    constructor(IEntryPoint _ep) payable {
+        ep = _ep;
+    }
+
+    function validateUserOp(UserOperation calldata userOp, bytes32, address, uint256 missingAccountFunds)
+    external override returns (uint256 sigTimeRange) {
+        ep.depositTo{value : missingAccountFunds}(address(this));
+        if (userOp.nonce == 1) {
+            this.touchStorage{gas: 1000}();
+        } else if (userOp.nonce == 2) {
+            address paymaster = address(bytes20(userOp.paymasterAndData[: 20]));
+            this.touchPaymaster{gas: 1000}(paymaster);
```

Comment:
add a comment such as: "this method call can only succeed of paymaster's code is already warm (e.g. in simulation, which reference it before validation, while in execution it is only referenced afterwards."

Reply:
Added comments